File: SQLiteRepository.Test/UnitTest1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SQLiteRepository.SQLiteQuery;

namespace SQLiteRepository.Test
{
    [TestClass]
    public class UnitTest1
    {
        private string _path = @"C:\Users\Jonathan\Documents\SignWriter Studio Sample Files\LESHO\LESHO - Copy (7).SWS";

        [TestMethod]
        public void GetData()
        {
           

            var selectColumns = new List<string> { "IDDictionary" };

            var query = new GetQuery
            {
                Path = _path,
                TableName = "Dictionary",
                Columns = selectColumns
            };
            var result = GetSqlite.GetData(query);

            var firstOrDefault = result.TabularResults.FirstOrDefault();
            var actual = 0;
            if (firstOrDefault != null)
            {
                actual = firstOrDefault.Count;
            }

            Assert.AreEqual(1279, actual);
        }

        [TestMethod]
        public void InsertData()
        {
            
            var insertColumns = new List<string> { "IDDictionary", "IDSignLanguage", "GUID" };

            var values = new List<List<string>>
            {
                new List< string> {"112555" , "1" , Guid.NewGuid().ToString()} ,
                new List< string> {"112556" , "1" , Guid.NewGuid().ToString()} ,
                new List< string> {"112557" , "1" , Guid.NewGuid().ToString()}  
            };

            var query = new InsertQuery
            {
                Path = _path,
                TableName = "Dictionary",
                Columns = insertColumns,
                Values = values
            };
            var result = query.Execute();

            Assert.AreEqual(3, result.AffectedRows);
        }

        [TestMethod]
        public void UpdateData()
        {
            
            var updateColumns = new List<string> { "IDDictionary", "IDSignLanguage", "GUID" };

            var newValues = new List<List<object>>
            {
                new List< object> {112555 , 1 , Guid.NewGuid()} ,
                new List< object> {112556 , 1 , Guid.NewGuid()} ,
                new List< object> {112557 , 1 , Guid.NewGuid()}  
            };

            var query = new UpdateQuery
            {
                Path = _path,
                TableName = "Dictionary",
                Columns = updateColumns,
                Values = newValues
            };
            var result = query.Execute();

            Assert.AreEqual(3, result.AffectedRows);
        }

        [TestMethod]
        public void DeleteData()
        {
            
            var where = "[IDDictionary] = '112557'";

            var query = new DeleteQuery
            {
                Path = _path,
                TableName = "Dictionary",
                Where = where
            };
            var result = query.Execute();

            Assert.AreEqual(1, result.AffectedRows);
        }

        [TestMethod]
        public void DeleteDataWhereIn()
        {
            
            var wherein = Tuple.Create("IDDictionary", new List<string> { "112555", "112556", "112557" });


            var query = new DeleteQuery
            {
                Path = _path,
                TableName = "Dictionary",
                WhereIn = wherein
            };

            var result = query.Execute();
            Assert.AreEqual(3, result.AffectedRows);
        }

        [TestMethod]
        public void UnitOfWork()
        {

            var insertColumns = new List<string> { "IDDictionary", "IDSignLanguage", "GUID" };

            var values = new List<List<string>>
            {
                new List< string> {"112555" , "1" , Guid.NewGuid().ToString()} ,
                new List< string> {"112556" , "1" , Guid.NewGuid().ToString()} ,
                new List< string> {"112557" , "1" , Guid.NewGuid().ToString()}  
            };

            var insertquery = new InsertQuery
            {
                Path = _path,
                TableName = "Dictionary",
                Columns = insertColumns,
                Values = values
            };

            var wherein = Tuple.Create("IDDictionary", new List<string> { "112555", "112556", "112557" });

            var updateColumns = new List<string> { "IDDictionary", "IDSignLanguage", "GUID" };

            var newValues = new List<List<object>>
            {
                new List< object> {112555 , 1 , Guid.NewGuid()} ,
                new List< object> {112556 , 1 , Guid.NewGuid()} ,
                new List< object> {112557 , 1 , Guid.NewGuid()}  
            };

            var updatequery = new UpdateQuery
            {
                Path = _path,
                TableName = "Dictionary",
                Columns = updateColumns,
                Values = newValues
            };
            var deletequery = new DeleteQuery
            {
                Path = _path,
                TableName = "Dictionary",
                WhereIn = wherein
            };
            var selectColumns = new List<string> { "IDDictionary" };

            var getquery = new GetQuery
            {
                Path = _path,
                TableName = "Dictionary",
                Columns = selectColumns
            };
            var unitofWork = new UnitOfWork();
           
            unitofWork.Queries.Add(insertquery);
            unitofWork.Queries.Add(updatequery);
            unitofWork.Queries.Add(deletequery);
            unitofWork.Queries.Add(getquery);

            var result = unitofWork.Execute();

            Assert.AreEqual(4, result.Count());
        }
    }
}

File: SQLiteAdapters.Test/UnitTest1.vb
Imports System.Collections.Generic
Imports System.Linq
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports SignWriterStudio.SQLiteAdapters

<TestClass> _
Public Class UnitTest1
    Private Const Path As String = "C:\Users\Jonathan\Documents\SignWriter Studio Sample Files\LESHO\LESHO - Copy (9) - Copy.SWS"

    <TestMethod> _
    Public Sub GetData()


        Dim selectColumns = New List(Of String)() From { _
            "IDDictionary" _
        }

        Dim query = New GetQuery() With { _
            .Path = Path, _
            .TableName = "Dictionary", _
            .Columns = selectColumns _
        }
        Dim result = GetSqlite.GetData(query)

        Dim firstOrDefault = result.TabularResults.FirstOrDefault()
        Dim actual = 0
        If firstOrDefault IsNot Nothing Then
            actual = firstOrDefault.Count
        End If

        Assert.AreEqual(1279, actual)
    End Sub

    <TestMethod> _
    Public Sub InsertData()

        Dim insertColumns = New List(Of String)() From { _
            "IDDictionary", _
            "IDSignLanguage", _
            "GUID" _
        }

        Dim values = New List(Of List(Of String))() From { _
            New List(Of String)() From { _
                "112555", _
                "1", _
                Guid.NewGuid().ToString() _
            }, _
            New List(Of String)() From { _
                "112556", _
                "1", _
                Guid.NewGuid().ToString() _
            }, _
            New List(Of String)() From { _
                "112557", _
                "1", _
                Guid.NewGuid().ToString() _
            } _
        }

        Dim query = New InsertQuery() With { _
            .Path = Path, _
            .TableName = "Dictionary", _
            .Columns = insertColumns, _
            .Values = values _
        }
        Dim result = query.Execute()

        Assert.AreEqual(3, result.AffectedRows)
    End Sub

    <TestMethod> _
    Public Sub UpdateData()

        Dim updateColumns = New List(Of String)() From { _
            "IDDictionary", _
            "IDSignLanguage", _
            "GUID" _
        }

        Dim newValues = New List(Of List(Of String))() From { _
            New List(Of String)() From {"112555", "1", Guid.NewGuid().ToString()}, _
            New List(Of String)() From {"112556", "1", Guid.NewGuid().ToString()}, _
            New List(Of String)() From {"112557", "1", Guid.NewGuid().ToString()}}

        Dim query = New UpdateQuery() With { _
            .Path = Path, _
            .TableName = "Dictionary", _
            .Columns = updateColumns, _
            .Values = newValues _
        }
        Dim result = query.Execute()

        Assert.AreEqual(3, result.AffectedRows)
    End Sub

    <TestMethod> _
    Public Sub DeleteData()

        Const where As String = "[IDDictionary] = '112557'"

        Dim query = New DeleteQuery() With { _
            .Path = Path, _
            .TableName = "Dictionary", _
            .Where = where _
        }
        Dim result = query.Execute()

        Assert.AreEqual(1, result.AffectedRows)
    End Sub

    <TestMethod> _
    Public Sub DeleteDataWhereIn()

        Dim wherein = Tuple.Create("IDDictionary", New List(Of String)() From { _
            "112555", "112556", "112557"})


        Dim query = New DeleteQuery() With { _
            .Path = Path, _
            .TableName = "Dictionary", _
            .WhereIn = wherein _
        }

        Dim result = query.Execute()
        Assert.AreEqual(3, result.AffectedRows)
    End Sub

    <TestMethod> _
    Public Sub UnitOfWork()

        Dim insertColumns = New List(Of String)() From { _
            "IDDictionary", _
            "IDSignLanguage", _
            "GUID" _
        }

        Dim values = New List(Of List(Of String))() From { _
            New List(Of String)() From {"112555", "1", Guid.NewGuid().ToString()}, _
            New List(Of String)() From {"112556", "1", Guid.NewGuid().ToString()}, _
            New List(Of String)() From {"112557", "1", Guid.NewGuid().ToString()}}

        Dim insertquery = New InsertQuery() With { _
            .Path = Path, _
            .TableName = "Dictionary", _
            .Columns = insertColumns, _
            .Values = values _
        }

        Dim wherein = Tuple.Create("IDDictionary", New List(Of String)() From { _
            "112555", _
            "112556", _
            "112557" _
        })

        Dim updateColumns = New List(Of String)() From { _
            "IDDictionary", _
            "IDSignLanguage", _
            "GUID" _
        }

        Dim newValues = New List(Of List(Of String))() From { _
            New List(Of String)() From {"112555", "1", Guid.NewGuid().ToString()}, _
            New List(Of String)() From {"112556", "1", Guid.NewGuid().ToString()}, _
            New List(Of String)() From {"112557", "1", Guid.NewGuid().ToString()} _
        }

        Dim updatequery = New UpdateQuery() With { _
            .Path = Path, _
            .TableName = "Dictionary", _
            .Columns = updateColumns, _
            .Values = newValues _
        }
        Dim deletequery = New DeleteQuery() With { _
            .Path = Path, _
            .TableName = "Dictionary", _
            .WhereIn = wherein _
        }
        Dim selectColumns = New List(Of String)() From { _
            "IDDictionary" _
        }

        Dim getquery = New GetQuery() With { _
            .Path = Path, _
            .TableName = "Dictionary", _
            .Columns = selectColumns _
        }
        Dim unitofWork1 = New UnitOfWork()

        unitofWork1.Queries.Add(insertquery)
        unitofWork1.Queries.Add(updatequery)
        unitofWork1.Queries.Add(deletequery)
        unitofWork1.Queries.Add(getquery)

        Dim result = unitofWork1.Execute()

        Assert.AreEqual(4, result.Count())
    End Sub
End Class
